src/token-group/__tests__/generic-token-group.test.tsx

```diff
@@ -0,0 +1,40 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import GenericTokenGroup, { GenericTokenGroupProps } from '../../../lib/components/token-group/generic-token-group';
+import createWrapper, { TokenGroupWrapper } from '../../../lib/components/test-utils/dom';
+
+interface Item {
+  name: string;
+}
+
+const defaultProps: GenericTokenGroupProps<Item> = {
+  items: generateItems(5),
+  alignment: 'horizontal',
+  renderItem: item => item.name,
+  getItemAttributes: item => ({ name: item.name }),
+};
+
+function renderTokenGroup(props: Partial<GenericTokenGroupProps<Item>> = {}): TokenGroupWrapper {
+  const { container } = render(<GenericTokenGroup {...defaultProps} {...props} />);
+  return createWrapper(container).findTokenGroup()!;
+}
+
+function generateItems(count: number) {
+  return [...new Array(count)].map((_, index) => ({ name: `name-${index}` }));
+}
+
+describe('GenericTokenGroup', () => {
+  test('tokens have role="group" and aria-label set', async () => {
+    renderTokenGroup();
+
+    await expect(screen.findByRole('group', { name: 'name-0' })).resolves.toBeDefined();
+  });
+
+  test('tokens do not have role and aria-label set when `asList` is used', async () => {
```

Comment:
I'm not convinced by this logic - I don't think that `listitem` gives the same semantics/benefits as `group`, so I think we should use both

Reply:
What is the difference? From testing it seems that it works just the same: https://d21d5uik3ws71m.cloudfront.net/components/7b802919d9f9e3d708cb24f9dab54e63bd408a67/index.html#/light/token-group/generic

We also do not use explicit groups in e.g. multiselect.

Reply:
Keeping the group adds another level of nesting

Reply:
When navigating the token it enters and exits the group twice if we keep the role="group" when also using a list.

https://user-images.githubusercontent.com/20790937/227233539-62898fac-00e6-480d-8c16-55ef24e6a1b5.mov

Reply:
got it, it sounds like the issue here is more VoiceOver